### PR TITLE
Update Xunit to 2.4.2

### PR DESCRIPTION
**Pull request by ericstj**

### CODE REVIEW

**Review by ViktorHofer — APPROVED** [review]

**Comment by ViktorHofer** [comment]
@dotnet/dnceng can you please help with diagnosing the test failures? I spent some time looking at the binlogs to see if the work items are submitted incorrectly but as the XUnitWorkItems feature is something very unfamiliar to me, I don't have any clues.

All work items fail with the following error:
```
✖DownloadError: Unable to download https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/1a5f89f6-d8da-4080-b15f-242650c914a8/nuget/v3/flat2/xunit.runner.console/2.4.2/xunit.runner.console.2.4.2.nupkg after retrying. Execution may fail as a result.
```

**Comment by dkurepa** [comment]
Looking at the logs, I think it's failing on trying to get `xunit.runner.console 2.4.2` from the [dotnet-eng](https://dev.azure.com/dnceng/public/_artifacts/feed/dotnet-eng), but it's not available, I'm only seeing `2.4.2-pre**` packages. Should there be a `xunit.runner.console 2.4.2` in there too?

**Comment by ViktorHofer** [comment]
Yes, it was just recently published to nuget.org (two weeks ago). Can you please add the package to the dotnet-eng feed?

**Comment by dkurepa** [comment]
We usually add packages like this to the dotnet-public feed. I just checked and it's already there. Is it possible to fetch it from there?

**Comment by ViktorHofer** [comment]
If the prerelease versions were pushed to dotnet-eng as well, the we should also push the stable version there.

For the future we can for sure avoid having these in dotnet-eng.

**Comment by dkurepa** [comment]
I added 2.4.2 to dotnet-eng, should be unblocked now

**Review by ViktorHofer — DISMISSED** [review]

File: src/Microsoft.DotNet.Helix/Sdk/Readme.md [hunk 1]
```diff
@@ -228,7 +228,7 @@ Given a local folder `$(TestFolder)` containing `runtests.cmd`, this will run `r
     <!-- TargetFramework of the xunit.runner.dll to use when running the tests -->
     <XUnitRuntimeTargetFramework>netcoreapp2.0</XUnitRuntimeTargetFramework>
     <!-- PackageVersion of xunit.runner.console to use -->
-    <XUnitRunnerVersion>2.4.2-pre.9</XUnitRunnerVersion>
+    <XUnitRunnerVersion>2.4.2/XUnitRunnerVersion>
```
Comment (MattGal) [inline src/Microsoft.DotNet.Helix/Sdk/Readme.md:231]: Typo: Deleted a >

**Review by MattGal — CHANGES_REQUESTED** [review]
Should fix the typo before merging.

File: src/Microsoft.DotNet.Helix/Sdk/Readme.md (hunk 1 above)
Comment (ViktorHofer) [reply src/Microsoft.DotNet.Helix/Sdk/Readme.md]: Thanks, fixed the typo.

**Review by MattGal — APPROVED** [review]

**Review by ViktorHofer — APPROVED** [review]

**Review by dougbu — APPROVED** [review]

**Pull request merged**